source/vibe/textfilter/urlencode.d

```diff
@@ -40,12 +42,12 @@ string urlEncode(string str, string allowed_chars = null)
 	return str;
 }
 
-unittest {
+@safe unittest {
 	string s = "hello-world";
 	assert(s.urlEncode().ptr == s.ptr);
 }
 
-private bool isCorrectHexNum(string str)
+private auto isCorrectHexNum(const(char)[] str)
```

Comment:
Ugh okay, that's rather ugly having to obfuscate the return value just to get attribute inference. It's just a private function, so it's not that important, though.